tensorflow/python/eager/function.py

```diff
@@ -105,9 +105,17 @@ def _make_input_signature_hashable(elem):
   Returns:
     A hashable object for the requested input signature
   """
+  try:
+    hash(elem)
```

Comment:
I think if we do return the value of hash, it could be faster. 

try:
    v = hash(elem)
  except TypeError:
    return _make_input_signature_hashable_helper(elem)
return v

Reply:
To clarify, if a tuple is input then the output would be a tuple of the hashes of the elements right? So later when the cache key is computed, it would be hashing some hashes

Making this modification without trying to modify the Exception case so that hashes of those return values are also returned breaks some test cases. Should I look into returning the hashes in the Exception case too?